test/hotspot/jtreg/compiler/vectorization/runner/ArrayShiftOpTest.java

```diff
@@ -68,6 +70,8 @@ public ArrayShiftOpTest() {
     }
 
     @Test
+    @IR(applyIfCPUFeatureOr = {"asimd", "true", "avx2", "true"},
+        counts = {IRNode.STORE_VECTOR, ">0"})
     public int[] intCombinedRotateShift() {
         int[] res = new int[SIZE];
         for (int i = 0; i < SIZE; i++) {
```

Comment:
IR checking is currently being done for STORE node where as the test is written for ROTATE operation.  AVX512F has direct rotate instruction support. Same comment applies to other cases, we should be checking for the exact operation nodes.

Reply:
Thanks for your good suggestion! In my latest commit, IR rules are updated with more exact operations in most of the cases. The ROTATE operation applies for x86 only so I added an additional rule for `avx512f` there.